conbench/app/_plots.py

```diff
@@ -35,22 +38,19 @@ def get_outliers(self, benchmarks):
         return outliers, outlier_ids, outlier_names
 
     def get_history_plot(self, benchmark, run, i=0):
-        history = self._get_history(benchmark)
-        if history:
+        samples = get_history_for_benchmark(benchmark_result_id=benchmark["id"])
+
+        if len(samples) > 0:
+            assert isinstance(samples[0], HistorySample)
+
             return json.dumps(
                 bokeh.embed.json_item(
-                    time_series_plot(history, benchmark, run),
+                    time_series_plot(samples, benchmark, run),
                     f"plot-history-{i}",
                 )
             )
-        return None
 
-    def _get_history(self, benchmark):
-        response = self.api_get("api.history", benchmark_id=benchmark["id"])
-        if response.status_code != 200:
-            self.flash("Error getting history.")
-            return []
-        return response.json
```

Comment:
this removes the API serialization/deserialization indirection described in https://github.com/conbench/conbench/issues/796.

Reply:
Thank you for this! This is a huge improvement IMO